rct_image_tools/src/rct_image_tools/modified_circle_grid_finder.cpp

```diff
@@ -106,38 +101,94 @@ static std::vector<cv::Point2d> extractKeyPoints(const cv::Mat& image, const std
   std::size_t temp_rows = flipped ? cols : rows;
   std::size_t temp_cols = flipped ? rows : cols;
 
+  std::size_t start_first_row = 0;
+  std::size_t end_first_row = cols - 1;
+  std::size_t start_last_row = rows * cols - cols;
+  std::size_t end_last_row = rows * cols - 1;
+
   // Determine which circle is the largest
   double start_first_row_size = -1.0;
   double start_last_row_size = -1.0;
   double end_first_row_size = -1.0;
   double end_last_row_size = -1.0;
 
+  // Determine size of corners relative to their adjacent circles
+  double start_first_row_avg_rel_size = 0;
+  double start_last_row_avg_rel_size = 0;
+  double end_first_row_avg_rel_size = 0;
+  double end_last_row_avg_rel_size = 0;
+
   cv::Point2d start_last_row_pt = centers[start_last_row];
   cv::Point2d end_last_row_pt = centers[end_last_row];
   cv::Point2d start_first_row_pt = centers[start_first_row];
   cv::Point2d end_first_row_pt = centers[end_first_row];
 
+  // Get the size of every circle
+  std::vector<double> center_sizes(centers.size(), 1.0);
+
   for (std::size_t i = 0; i < keypoints.size(); i++)
   {
     double x = keypoints[i].pt.x;
     double y = keypoints[i].pt.y;
     double ksize = keypoints[i].size;
 
+    for (std::size_t j = 0; j < centers.size(); j++)
+    {
+      auto center = centers[j];
```

Comment:
Don't copy unnecessarily
```suggestion
      const auto& center = centers[j];
```